File: Topic Based Problems/01 Fundamentals/Beginner/16 Swap Two Numbers/Python/SwapTwoNumbersUsingTemp.py
# SwapTwoNumbersUsingTemp.py
# Swap two numbers in Python using temp variable
# Version: 1.0
# Date: 2023-09-24
# Author: SharifdotG (sharifmdyousuf007@gmail.com)

firstNumber = int(input("Enter first number: "))
secondNumber = int(input("Enter second number: "))

print("Before swapping:")
print("First number:", firstNumber)
print("Second number:", secondNumber)

firstNumber, secondNumber = secondNumber, firstNumber

print("After swapping:")
print("First number:", firstNumber)
print("Second number:", secondNumber)

File: Topic Based Problems/01 Fundamentals/01 Beginner/16 Swap Two Numbers/C/SwapTwoNumbersUsingTemp.c
/**
 * @file SwapTwoNumbersUsingTemp.c
 * @author SharifdotG (sharifmdyousuf007@gmail.com)
 * @brief Swap Two Numbers in C using Temp Variable
 * @version 1.0
 * @date 2023-09-24
 * 
 * @copyright Copyright (c) 2023
 * 
 */

#include <stdio.h>

int main() {
    int firstNumber, secondNumber, temp;

    printf("Enter first number: ");
    scanf("%d", &firstNumber);

    printf("Enter second number: ");
    scanf("%d", &secondNumber);

    printf("Before swapping: \n");
    printf("First number: %d\n", firstNumber);
    printf("Second number: %d\n", secondNumber);

    temp = firstNumber;
    firstNumber = secondNumber;
    secondNumber = temp;

    printf("After swapping: \n");
    printf("First number: %d\n", firstNumber);
    printf("Second number: %d\n", secondNumber);

    return 0;
}
